modules/ml/src/main/java/org/apache/ignite/ml/svm/SVMLinearBinaryClassificationTrainer.java

```diff
@@ -32,33 +35,50 @@
  * and +1 labels for two classes and makes binary classification. </p> The paper about this algorithm could be found
  * here https://arxiv.org/abs/1409.1458.
  */
-public class SVMLinearBinaryClassificationTrainer implements Trainer<SVMLinearBinaryClassificationModel, LabeledDataset> {
+public class SVMLinearBinaryClassificationTrainer<K, V>
+    implements DatasetTrainer<K, V, SVMLinearBinaryClassificationModel> {
     /** Amount of outer SDCA algorithm iterations. */
-    private int amountOfIterations = 20;
+    private int amountOfIterations = 200;
 
     /** Amount of local SDCA algorithm iterations. */
-    private int amountOfLocIterations = 50;
+    private int amountOfLocIterations = 100;
 
     /** Regularization parameter. */
-    private double lambda = 0.2;
+    private double lambda = 0.4;
 
-    /** This flag enables distributed mode for this algorithm. */
-    private boolean isDistributed;
+    /** Dataset. */
+    private Dataset<SVMPartitionContext, LabeledDataset<Double, LabeledVector>> dataset;
```

Comment:
Would be better to make `dataset` a variable within the `fit` method (not a class field). And close if immediately after fitting as well.